src/Microsoft.Health.Fhir.Api/Features/BackgroundJobService/JobFactory.cs

```diff
@@ -47,16 +52,36 @@ public JobFactory(IEnumerable<Func<IJob>> jobFactories, ILogger<JobFactory> logg
             }
         }
 
-        public IJob Create(JobInfo jobInfo)
+        public IScoped<IJob> Create(JobInfo jobInfo)
         {
             EnsureArg.IsNotNull(jobInfo, nameof(jobInfo));
 
-            if (_jobFactoryLookup.TryGetValue(jobInfo.GetJobTypeId() ?? int.MinValue, out Func<IJob> jobFactory))
+            IScoped<IEnumerable<Func<IJob>>> scope = _jobFactories.Invoke();
```

Comment:
Nit: should this be inside the if statement to avoid making a scope when it isn't needed?